google/cloud/spanner/connection.h

```diff
@@ -54,9 +55,27 @@ class Connection {
     KeySet keys;
     std::vector<std::string> columns;
     ReadOptions read_options;
+
+    // TODO(#307): Refactor once correct location for session implemented.
+    google::cloud::optional<std::string> session_name;
+
+    ReadParams(Transaction transaction, std::string table, KeySet keys,
+               std::vector<std::string> columns, ReadOptions read_options,
+               google::cloud::optional<std::string> session_name = {})
+        : transaction(std::move(transaction)),
+          table(std::move(table)),
+          keys(std::move(keys)),
+          columns(std::move(columns)),
+          read_options(std::move(read_options)),
+          session_name(std::move(session_name)) {}
   };
   virtual StatusOr<ResultSet> Read(ReadParams) = 0;
 
+  virtual StatusOr<ResultSet> Read(ReadPartition) = 0;
```

Comment:
I don't think we want a Read() overload here. `Client::Read(ReadPartition)` should call `Connection::Read(ReadParams)`. Do you think that'll work?

Reply:
I was under the impression we added the Connection interface to allow customers to mock calls. It seemed useful to expose a different Read signature that was intended to be used with the results of PartitionRead.

Reply:
Overload removed. client will pass a ReadParams to Connection::Read.